src/Silex/Application.php

```diff
@@ -454,8 +454,8 @@ public function mount($prefix, $controllers)
             }
 
             $controllers = $connectedControllers;
-        } elseif (!$controllers instanceof ControllerCollection) {
-            throw new \LogicException('The "mount" method takes either a "ControllerCollection" or a "ControllerProviderInterface" instance.');
+        } elseif (!$controllers instanceof ControllerCollection && !is_callable($controllers)) {
+            throw new \LogicException('The "mount" method takes either a "ControllerCollection" instance, "ControllerProviderInterface" instance, or a callable.');
```

Comment:
`"ControllerProviderInterface" instance` I get what you try to state but it is impossible to have an instance of an interface.
How about
```php
sprintf('The "mount" method takes either a "ControllerCollection", "ControllerProviderInterface" implementation, or `callable`. Got: "%s".', is_object($controllers) ? get_class($controllers) : gettype($controllers)));
```